### PR TITLE
delete parameters from context.json_data before validating schema

### LINKED ISSUE #22: Getting error_code 500 - "wrong json schema"
Please help me with the following issue.

I am using cangaroo to integrate "wombat/amazon_integration" into my spree app. Everything seems to working as expected but I am getting the below error in attached in the "Interactor::Context" json.
`{message="wrong json schema", error_code=500}`

The problem seems to be with the Schema of the json "ValidateJsonSchema" is expecting, but my response from the "wombat/amazon_integration" is as expected and as per the guidelines given by wombat. Please find the "req.parsed_response" below.

`{"request_id"=>"c5824270-beda-4c79-8248-2bebda3ede31",
 "summary"=>"Successfully received 7 order(s) and 0 shipment(s) from Amazon MWS.",
 "parameters"=>{"amazon_orders_last_polling_datetime"=>"2016-04-18T16:09:12Z"},
 "orders"=>
  [{"id"=>"404-1821001-3722760",
    "number"=>"404-1821001-3722760",
    "channel"=>"Amazon.in",
    "fulfillment_channel"=>"MFN",
    "currency"=>"INR",
    "status"=>"Unshipped",
    "placed_on"=>"2016-04-13T14:15:48Z",
    "updated_at"=>"2016-04-13T14:46:05Z",
    "email"=>"7qx1c2fn6d1rcje@marketplace.amazon.in",
    "totals"=>
     {"item"=>14999.0,
      "adjustment"=>0.0,
      "tax"=>0.0,
      "shipping"=>0.0,
      "order"=>14999.0,
      "payment"=>14999.0},
    "adjustments"=>
     [{"name"=>"Shipping Discount", "value"=>0.0},
      {"name"=>"Promotion Discount", "value"=>0.0},
      {"name"=>"Amazon Tax", "value"=>0.0},
      {"name"=>"Gift Wrap Price", "value"=>0.0},
      {"name"=>"Gift Wrap Tax", "value"=>0.0}],
    "line_items"=>
     [{"name"=>"Prague Coffee Table - Mahogany",
       "price"=>14999.0,
       "product_id"=>"FRTBCF11MH10001",
       "quantity"=>1,
       "asin"=>"B0188P7GP0",
       "options"=>{}}],
    "payments"=>[{"amount"=>14999.0, "payment_method"=>"Amazon", "status"=>"complete"}],
    "shipping_address"=>
     {"firstname"=>"Rohit",
      "lastname"=>"Agarwal",
      "address1"=>"Salasar Aangan, Nr. Prathmesh Ashish,",
      "address2"=>"Off Kanakia Road, Mira Road EAST",
      "city"=>"THANE",
      "zipcode"=>"401107",
      "phone"=>"9876543210",
      "country"=>"IN",
      "state"=>"MAHARASHTRA"},
    "billing_address"=>
     {"firstname"=>"Rohit",
      "lastname"=>"Agarwal",
      "address1"=>"Salasar Aangan, Nr. Prathmesh Ashish,",
      "address2"=>"Off Kanakia Road, Mira Road EAST",
      "city"=>"THANE",
      "zipcode"=>"401107",
      "phone"=>"9876543210",
      "country"=>"IN",
      "state"=>"MAHARASHTRA"},
    "amazon_shipping_method"=>"Expedited"}]
}`

Please let me know what am I doing wrong ?

### PR DESCRIPTION
**Pull request by rohitnick**
`context.json_body` can also include `parameters` in addition to `request_id` and `summary`, which is as per the guidelines given by wombat.

Therefore removing it from `context.json_body` before validating objects schema.

fixes #22

### CODE REVIEW

**Comment by iloveitaly** [comment]
@rohitnick What do you think about adding these parameters to the JSON schema as optional fields? Was there a reason that wouldn't work?

Thanks for your work here! Exciting to see this project pick up steam!

**Comment by rohitnick** [comment]
@iloveitaly Essentially what you are saying will definitely work and should be the right way of doing things. Adding the additional parameters to the JSON schema as optional fields will also solve this issue.

But I was in a hurry and didn't wanted to go through the 'json-schema' so fixed the issue by removing the parameters.

**Comment by bricesanchez** [comment]
This PR helped me to go forward with my chaindrive_integration but i had a JSON Validator problem until i've made this change : 

```git
# app/jobs/cangaroo/poll_job.rb
@@ -30,7 +30,7 @@ module Cangaroo
      command = HandleRequest.call(
        key: destination_connection.key,
        token: destination_connection.token,
        json_body: response.to_json,
        json_body: response,
        jobs: Rails.configuration.cangaroo.jobs
      )
```

File: app/interactors/cangaroo/count_json_object.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def call
     private
 
     def prepare_context
-      context.data = JSON.parse(context.json_body)
+      context.data = context.json_body
```
Comment (AlessioRocco) [inline app/interactors/cangaroo/count_json_object.rb]: @rohitnick why did you change this, also on other files?

**Comment by AlessioRocco** [comment]
@rohitnick can you rebase master and make specs pass? If you don't want do it let me know and I'll fix it.

**Comment by rohitnick** [comment]
@AlessioRocco : I can fix this but it can take me upto 48 hrs, as I am busy with other stuff. If you fix it faster, please do. :)

**Comment by AlessioRocco** [comment]
@rohitnick I had some time today and I've fixed it with this [PR](https://github.com/nebulab/cangaroo/pull/32), I've stolen the solution from you so thanks for your work.
